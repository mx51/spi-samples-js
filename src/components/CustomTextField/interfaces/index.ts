import { IFormEventValue } from '../../PairPage/PairForm/interfaces';

type Partial<T> = {
  [P in keyof T]?: T[P];
};

type InputProps = {
  endAdornment?: React.ReactNode;
};

type TInputProps = {
  maxLength?: number;
};

export interface ICustomTextField {
  disabled?: boolean;
  error?: boolean;
  fullWidth?: boolean;
  helperText?: string;
  id?: string;
  inputProps?: Partial<TInputProps>;
  InputProps?: Partial<InputProps>;
  label?: string;
  margin?: 'dense' | 'none';
  onBlur?: (event: IFormEventValue) => void;
  onChange?: (event: IFormEventValue) => void;
  required?: boolean;
  value?: string;
  variant?: 'filled' | 'standard' | 'outlined' | undefined;
}

export interface IErrorInputAdornment {
  isValid: boolean;
}
