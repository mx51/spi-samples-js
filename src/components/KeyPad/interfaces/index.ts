export interface IKeyPadProps {
  title: string;
  subtitle: string;
  defaultAmount: number;
  onAmountChange: (amount: number) => void;
  open: boolean;
  onClose: () => void;
}
