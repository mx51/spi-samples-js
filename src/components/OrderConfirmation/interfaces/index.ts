export interface IOrderConfirmation {
  title: string;
  pathname: string;
  editSubtotal: boolean;
}

export interface ITitleStrategy {
  Pay: string;
  Cashout: string;
  Refund: string;
}
