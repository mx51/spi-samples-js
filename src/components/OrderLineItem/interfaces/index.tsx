export interface IOrderLineItemProps {
  label: string;
  amount: number;
}
