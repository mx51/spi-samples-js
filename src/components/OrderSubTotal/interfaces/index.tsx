export interface IOrderSubTotalProps {
  label: string;
  amount: number | null;
}
