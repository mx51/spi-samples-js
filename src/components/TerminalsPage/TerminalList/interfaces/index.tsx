export interface ITerminalRecord {
  posId: string;
  pairingStatus: PairingStatus;
  deviceAddress: string;
  serialNumber: string;
  status: string;
}
export interface ITerminalsRecords {
  terminals: Array<ITerminalRecord>;
}

export enum PairingStatus {
  Connecting,
  Connected,
  Disconnected,
  Idle,
}
