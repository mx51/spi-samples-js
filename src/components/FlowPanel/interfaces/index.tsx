import { ITerminalProps } from '../../../redux/reducers/TerminalSlice/interfaces';

export interface IFlowPanel {
  flow: boolean;
  receiptToggle?: boolean;
}

export interface IFlowPanelState {
  flow: boolean;
  terminal?: ITerminalProps | null;
}
