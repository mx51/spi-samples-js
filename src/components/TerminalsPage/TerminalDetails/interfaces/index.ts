import React from 'react';
import { ITerminalProps } from '../../../../redux/reducers/TerminalSlice/interfaces';

export interface IReceiptToggles {
  settlement: boolean;
  settlementEnquiry: boolean;
}

export interface TabPanelProps {
  children: React.ReactNode;
  index: number;
  subtitle: string;
  title: string;
  value: number;
  receiptToggle?: IReceiptToggles;
  terminal?: ITerminalProps | null;
}

export interface ITerminalConfigurationConstants {
  title: string;
  content: string;
}

export interface IAboutTerminal {
  receiptToggle: IReceiptToggles;
  setReceiptToggle: (receiptToggle: IReceiptToggles) => void;
  terminal: ITerminalProps;
}
