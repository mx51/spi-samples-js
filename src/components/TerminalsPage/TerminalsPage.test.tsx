import React from 'react';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Terminals from '.';
import { defaultLocalIP } from '../../definitions/constants/spiConfigs';
import { IPairFormParams } from '../../redux/reducers/PairFormSlice/interfaces';
import { ITerminalState } from '../../redux/reducers/TerminalSlice/interfaces';
import mockWithRedux, { defaultMockPairFormParams, defaultMockTerminals } from '../../utils/tests/common';

function setupContainer(
  pairForm: IPairFormParams = defaultMockPairFormParams,
  terminals: ITerminalState = defaultMockTerminals
) {
  const customizedStore = {
    getState: () => ({
      common: {},
      pairForm,
      terminals,
    }),
    subscribe: jest.fn(),
    dispatch: jest.fn(),
  };

  return mockWithRedux(<Terminals />, customizedStore);
}

describe('Test <Terminals />', () => {
  let container: Any;
  afterEach(cleanup);

  beforeEach(() => {
    container = setupContainer();
  });

  test('should match TerminalsPage snapshot test', () => {
    // Assert
    expect(container).toMatchSnapshot();
  });

  test('should return a list of terminals', () => {
    // Arrange
    const terminalRecordKeywords: string[] = ['POS ID', 'Pairing status', 'EFTPOS address', 'Serial number'];

    // Assert
    for (let index = 0; index < terminalRecordKeywords.length; index += 1) {
      expect(container.innerHTML.includes(terminalRecordKeywords[index])).toBeTruthy();
    }
    expect(container.innerHTML.includes(defaultLocalIP)).toBeTruthy();
  });

  test('should return add pair terminals button when terminal list is empty', () => {
    // Arrange
    const mockContainer = mockWithRedux(<Terminals />);

    // Assert
    expect(mockContainer.innerHTML.includes('+ Pair terminal')).toBeTruthy();
    expect(mockContainer.innerHTML.includes(defaultLocalIP)).toBeFalsy();
  });
});
