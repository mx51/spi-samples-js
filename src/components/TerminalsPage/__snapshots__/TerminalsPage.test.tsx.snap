// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Terminals /> should match TerminalsPage snapshot test 1`] = `
<div>
  <div>
    <div
      class="makeStyles-root-4"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navbar-5 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <div
            class="MuiBox-root MuiBox-root-10 makeStyles-navbar-7"
          >
            <div
              class="MuiBox-root MuiBox-root-11"
            >
              <button
                aria-label="menu"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-8 MuiIconButton-colorInherit MuiIconButton-edgeStart"
                data-test-id="navbarMenuIcon"
                name="menuIcon"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <svg
                height="15"
                width="48"
              >
                PrimaryLogoIcon.svg
              </svg>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthMd"
    >
      <div
        class="MuiBox-root MuiBox-root-12"
      >
        <div
          class="MuiBox-root MuiBox-root-13"
        >
          <h1
            class="MuiTypography-root MuiTypography-h6"
          >
            Terminals
          </h1>
        </div>
        <div
          class="MuiBox-root MuiBox-root-14"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            href="/terminals/new"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              + Pair new terminal
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiTableContainer-root makeStyles-table-15 MuiPaper-elevation0 MuiPaper-rounded"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                POS ID
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Pairing status
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                EFTPOS address
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Serial number
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root makeStyles-unclickable-20"
              id="terminal_123-123-123"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
                scope="row"
              >
                test
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                <div
                  class="MuiChip-root makeStyles-chipUnpaired-18 MuiChip-sizeSmall"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall"
                  >
                    Unpaired
                  </span>
                </div>
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                192.168.1.10
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                123-123-123
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;
